File: node/src/internal/camera-wire.ts
import type { DetectionEvent } from '../camera/events.js';
import type { DetectionEventType } from '../camera/enums.js';

/**
 * Detection event message published via NATS.
 */
export interface DetectionEventMessage {
  /** Event lifecycle type */
  type: DetectionEventType;
  /** Full event data */
  data: DetectionEvent;
}

File: python/camera_ui_sdk/internal/camera_wire.py
from __future__ import annotations

from typing_extensions import TypedDict

from ..camera.enums import DetectionEventType
from ..camera.events import DetectionEvent


class DetectionEventMessage(TypedDict):
    """Detection event message published via NATS."""

    type: DetectionEventType
    """Event lifecycle type."""
    data: DetectionEvent
    """Full event data."""
